include/boost/geometry/algorithms/detail/envelope/box.hpp

```diff
@@ -40,36 +42,124 @@ namespace detail { namespace envelope
 {
 
 
-struct envelope_box_on_spheroid
+template
+<
+    std::size_t Index,
+    std::size_t Dimension,
+    std::size_t DimensionCount,
+    typename CS_Tag
+>
+struct envelope_indexed_box
 {
-    template<typename BoxIn, typename BoxOut>
+    template <typename BoxIn, typename BoxOut>
     static inline void apply(BoxIn const& box_in, BoxOut& mbr)
     {
-        BoxIn box_in_normalized = detail::return_normalized<BoxIn>(box_in);
+        detail::indexed_point_view<BoxIn const, Index> box_in_corner(box_in);
+        detail::indexed_point_view<BoxOut, Index> mbr_corner(mbr);
+
+        detail::conversion::point_to_point
+            <
+                detail::indexed_point_view<BoxIn const, Index>,
+                detail::indexed_point_view<BoxOut, Index>,
+                Dimension,
+                DimensionCount
+            >::apply(box_in_corner, mbr_corner);
+    }
+};
 
-        geometry::transform(box_in_normalized, mbr);
+template
+<
+    std::size_t Index,
+    std::size_t DimensionCount
+>
+struct envelope_indexed_box<Index, 0, DimensionCount, spherical_equatorial_tag>
+{
+    template <typename BoxIn, typename BoxOut>
+    static inline void apply(BoxIn const& box_in, BoxOut& mbr)
+    {
+
+       // transform() does not work with boxes of dimension higher
+        // than 2; to account for such boxes we transform the min/max
+        // points of the boxes using the indexed_point_view
+        detail::indexed_point_view<BoxIn const, Index> box_in_corner(box_in);
+        detail::indexed_point_view<BoxOut, Index> mbr_corner(mbr);
+
+        // first transform the units
+        transform_units(box_in_corner, mbr_corner);
```

Comment:
Yes, your understanding is correct.
A specialization for `Dimension` equal to `0` indeed transforms degrees/radians for the first 2 coordinates and does nothing for the remaining ones.
Yes, in theory you can instantiate it with `Dimension` equal to `1`. This is never done by the envelope code for non-Cartesian coordinate systems. If instantiated with `1`, then it behaves as if the coordinates where Cartesian.

I do not agree with your consistency argument. I do agree that it is not that common, but it does exist. We do have in several places a dimension range for which the algorithm/utility is implemented. It is the same here: there are two template parameters: `Dimension` and `DimensionCount`; the algorithm semantically deals with all coordinates within this dimension range.